packages/next/src/server/lib/router-utils/setup-dev-bundler.ts

```diff
@@ -549,30 +556,12 @@ async function startWatcher(opts: SetupOpts) {
       }
     }, 2)
 
-    function sendHmr(key: string, id: string, payload: HMR_ACTION_TYPES) {
-      // We've detected a change in some part of the graph. If nothing has
-      // been inserted into building yet, then this is the first change
-      // emitted, but their may be many more coming.
-      if (!hmrBuilding) {
-        hotReloader.send({ action: HMR_ACTIONS_SENT_TO_BROWSER.BUILDING })
-        hmrBuilding = true
-      }
+    function enqueueHmr(key: string, id: string, payload: HMR_ACTION_TYPES) {
       hmrPayloads.set(`${key}:${id}`, payload)
-      hmrEventHappened = true
-      sendHmrDebounce()
     }
 
-    function sendTurbopackMessage(payload: TurbopackUpdate) {
-      // We've detected a change in some part of the graph. If nothing has
-      // been inserted into building yet, then this is the first change
-      // emitted, but their may be many more coming.
-      if (!hmrBuilding) {
-        hotReloader.send({ action: HMR_ACTIONS_SENT_TO_BROWSER.BUILDING })
-        hmrBuilding = true
-      }
+    function enqueueTurbopackMessage(payload: TurbopackUpdate) {
       turbopackUpdates.push(payload)
-      hmrEventHappened = true
-      sendHmrDebounce()
```

Comment:
Why no longer calling this? Shouldn't it send out the events immediately and don't wait for the end event?

Reply:
What's the answer to this?

Reply:
I reintroduced this call, so it's updating immediately again.